pkg/client/delegation/client.go

```diff
@@ -0,0 +1,79 @@
+package delegation
+
+import (
+	"context"
+
+	"cosmossdk.io/depinject"
+
+	"github.com/pokt-network/poktroll/pkg/client"
+	"github.com/pokt-network/poktroll/pkg/client/events"
+)
+
+const (
+	// delegationEventQuery is the query used by the EventsQueryClient to subscribe
+	// to new delegation events from the the application module on chain.
+	// See: https://docs.cosmos.network/v0.47/learn/advanced/events#subscribing-to-events
+	// And: https://docs.cosmos.network/v0.47/learn/advanced/events#default-events
+	delegationEventQuery = "message.action='pocket.application.EventRedelegation'"
+	replayObsBufferSize  = 1 // the amount of events we want before they are emitted
```

Comment:
Similar feedback to https://github.com/pokt-network/poktroll/pull/220#discussion_r1417030309.